File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/services/CaregiversStatusService.java
package io.github.thunkier.thunkmetrc.wrapper.services;

import io.github.thunkier.thunkmetrc.client.MetrcClient;
import io.github.thunkier.thunkmetrc.wrapper.MetrcRateLimiter;import io.github.thunkier.thunkmetrc.wrapper.models.*;import java.util.List;

public class CaregiversStatusService {
    private final MetrcClient client;
    private final MetrcRateLimiter rateLimiter;

    public CaregiversStatusService(MetrcClient client, MetrcRateLimiter rateLimiter) {
        this.client = client;
        this.rateLimiter = rateLimiter;
    }

    /**
     * Retrieves the status of a Caregiver by their License Number for a specified Facility. Data returned by this endpoint is cached for up to one minute.
     * Permissions Required:
     * - Lookup Caregivers
     * GET /caregivers/v2/status/{caregiverLicenseNumber}
     * @param caregiverLicenseNumber Path parameter
     * @param licenseNumber Query parameter
     * @throws Exception If request fails
     * @return Response object
     */
    @SuppressWarnings("unchecked")
    public List<CaregiversStatus> getCaregiversStatusByCaregiverLicenseNumber(
        String caregiverLicenseNumber, String licenseNumber
    ) throws Exception {
        return rateLimiter.execute(null,true,
            () -> (List<CaregiversStatus>) client.caregiversStatus.getCaregiversStatusByCaregiverLicenseNumber(
                caregiverLicenseNumber, licenseNumber
            )
        );
    }

}

File: sdks/thunkmetrc-kotlin/wrapper/src/main/kotlin/io/github/thunkier/thunkmetrc/wrapper/services/CaregiversStatusService.kt
package io.github.thunkier.thunkmetrc.wrapper.services

import io.github.thunkier.thunkmetrc.client.MetrcClient
import io.github.thunkier.thunkmetrc.wrapper.MetrcRateLimiter
import io.github.thunkier.thunkmetrc.wrapper.MetrcExtensions
import io.github.thunkier.thunkmetrc.wrapper.models.*
import kotlinx.coroutines.flow.Flow

class CaregiversStatusService(private val client: MetrcClient, private val rateLimiter: MetrcRateLimiter) {

    /**
     * Retrieves the status of a Caregiver by their License Number for a specified Facility. Data returned by this endpoint is cached for up to one minute.
     * Permissions Required:
     * - Lookup Caregivers
     * GET /caregivers/v2/status/{caregiverLicenseNumber}
     * @param caregiverLicenseNumber Path parameter
     * @param licenseNumber Query parameter
     * @throws Exception If request fails
     * @return Response object
     */
    suspend fun getCaregiversStatusByCaregiverLicenseNumber(caregiverLicenseNumber: String, licenseNumber: String? = null): List<CaregiversStatus>? {
        return rateLimiter.execute(null,true,
        ) { 
            client.caregiversStatus.getCaregiversStatusByCaregiverLicenseNumber(caregiverLicenseNumber, licenseNumber, ) 
        } as? List<CaregiversStatus>
    }
}

